lms/djangoapps/courseware/student_field_overrides.py

```diff
@@ -0,0 +1,71 @@
+"""
+API related to providing field overrides for individual students.  This is used
+by the individual due dates feature.
+"""
+import json
+
+from .field_overrides import FieldOverrideProvider
+from .models import StudentFieldOverride
+
+
+class IndividualStudentOverrideProvider(FieldOverrideProvider):
+    """
+    A concrete implementation of
+    :class:`~courseware.field_overrides.FieldOverrideProvider` which allows for
+    overrides to be made on a per user basis.
+    """
+    def get(self, block, name, default):
+        return get_override_for_user(self.user, block, name, default)
+
+
+def get_override_for_user(user, block, name, default=None):
+    """
+    Gets the value of the overridden field for the `user`.  `block` and `name`
+    specify the block and the name of the field.  If the field is not
+    overridden for the given user, returns `default`.
+    """
+    try:
+        override = StudentFieldOverride.objects.get(
+            course_id=block.runtime.course_id,
+            location=block.location,
+            student_id=user.id,
+            field=name
+        )
+        field = block.fields[name]
```

Comment:
Does this mean there will be a separate query against `StudentFieldOverride` for potentially every block in the course (so hundreds of selects) when rendering the progress page? Can we prefetch all the overrides for a student in a course in one query when initializing the provider for this student/course?

Also, please use assertNumQueries() in tests to verify the query behavior.